### ISSUE TITLE
Contradicting information between, History balance and Transaction visualizer.

### ISSUE THREAD

**Issue by Transisto** [sparrowwallet/sparrow#602]
Transaction history show minus 0.95 BTC

Transaction viewer
0.3 sent
0.65 change

Wallet's conenction toggle is ON, but there's a red thing at the top.
Stuck on (connection has passed maxLifetime)
Anyway the transaction show as confirmed.
Just poiting this out that the wallet lose connection and can't reconnect by itself within ~12 hours.

Same type of problem as a previous issue, where the history isn't aware of it's own change addresses because maybe it's not watching it.

In other words the wallet history is aware of the utxo it spent but not the change it received.

**Comment by craigraw** [comment]
Definitely sounds like a server-related issue (as the red wallet icon indicates). If the server does not or cannot notify Sparrow that a change address has received a UTXO, Sparrow can't display it. Sparrow does try to explicitly request history for addresses it expects to see activity on (for example when you broadcast a transaction with change), but ultimately the server is the source of truth and Sparrow cannot show information it does not have.

We keep going round and round this same point - you point out an issue loading a transaction, and I reply that the server is at fault. I can't fix the server, so I'm not sure what you are expecting of me? I assure you Sparrow is watching all change addresses within the gap limit (or we would see widespread reports on this).

**Comment by Transisto** [comment]
Again, We never had any balance discrepancy/server issue on electrum in over 5 years of daily use? We're using the same server.

<img alt="image" src="https://user-images.githubusercontent.com/388716/177023388-6d5fb089-df0a-4168-9ddb-6dea9ee3845e.png">

<img alt="image" src="https://user-images.githubusercontent.com/388716/177023405-9c21ad04-ca22-4060-ad84-1009fa19bff0.png">

<img alt="image" src="https://user-images.githubusercontent.com/388716/177023433-8451a731-a3eb-4904-a8fe-d954d8770c3d.png">

There's a discrepancy between two location within Sparrow.

Sparrow History got told what by the server? That it received some money to one of it's change address but never spent the 0.65?

<img alt="image" src="https://user-images.githubusercontent.com/388716/177023504-62e6eb29-7a38-4aee-9af4-c53032939211.png">

**Comment by Transisto** [comment]
The reason why this is happening is that the 0.65 was stored on an earlier address that isn't monitored anymore.

The wallet still list that UTXO as being there although it's spent.

**Comment by craigraw** [comment]
Closing as a server (likely rate limiting) related issue, see #583.
